ecs-client.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const { ECSClient, StopTaskCommand } = require("@aws-sdk/client-ecs");
+const axios = require('axios');
+const callWithRetries = require('./call-with-retries');
+const Exception = require("./exception");
+const exitDeferred = require('./exit-deferred');
+
+const DEFAULT_TIMEOUT = 30000;
+const DEFAULT_REASON = 'ecs-client.js: default stop reason';
+
+class EcsClient {
+
+    /**
+     * @param {Object} options
+     * @param {Object} options.aws - AWS config
+     * @param {String} options.aws.accessKeyId
+     * @param {String} options.aws.secretAccessKey
+     * @param {String} options.aws.region
+     * @param {String} options.metadataUri
+     * @param {Number} [options.timeout=30000] - used for metadata fetch request
+     * @param {Object=} options.logger - An instance of sq-logger
+     */
+    constructor(options) {
+        if(options.logger == null){
+            throw new Exception(Exception.ErrorCode.ERROR_INVALID_PARAMETER, 'An sq-logger instance must be provided.');
+        }
+        this.logger = options.logger;
+        this.metadataUri = options.metadataUri;
+        this.client = new ECSClient(options.aws);
+        this.timeout = options.timeout || DEFAULT_TIMEOUT;
+    }
+
+    /**
+     * Trigger a stop task request to ECS
+     * @param reason
+     * @returns {Promise<*>}
+     */
+    async stopTask(reason) {
+        try {
+            this.logger.info(`ecs-client.js:: Stopping task...`);
+            const metadata = await callWithRetries(3, this, this._fetchMetadata);
+            return this._sendStopTaskCommand(metadata, reason);
+        } catch (e) {
+            this.logger.notify('ecs-client.js::stopTask').steps(0, 1).msg(`ecs-client.js:: Error stopping task: `, e);
+        }
+    }
+
+    /**
+     * Setup logger shutdown handler.
+     * Will be called on unhandledException / unhandledRejection
+     * @param exitOnErrorTimeout - Sets the delay before exiting process when unable to perform a successful stopTask request
+     */
+    setupShutdownHandler(exitOnErrorTimeout) {
```

Comment:
I'd rather call this method: **setupLoggerShutdownHandler** and add the logger as mandatory parameter (I would remove the logger parameter from constructor)

Reply:
Since we are using the logger when calling stopTask, the logger parameter in the constructor is necessary